local/modules/TheliaSmarty/Template/Plugins/UrlGenerator.php

```diff
@@ -119,6 +122,37 @@ public function generateUrlFunction($params, &$smarty)
                 $this->getArgsFromParam($params, array_merge(['noamp', 'path', 'file', 'target'], $excludeParams)),
                 $mode
             );
+            
+            $request = $this->getRequest();
+            $requestedLangCodeOrLocale = $params["lang"];
+            $view = $request->attributes->get('_view', null);
+            $viewId = $view === null ? null : $request->query->get($view . '_id', null);
+
+            if (null !== $requestedLangCodeOrLocale) {
+                if (strlen($requestedLangCodeOrLocale) > 2) {
+                    $lang = LangQuery::create()->findOneByLocale($requestedLangCodeOrLocale);
+                } else {
+                    $lang = LangQuery::create()->findOneByCode($requestedLangCodeOrLocale);
+                }
+
+                if (ConfigQuery::isMultiDomainActivated()) {
+                    $urlRewrite = RewritingUrlQuery::create()
+                        ->filterByView($view)
+                        ->filterByViewId($viewId)
+                        ->filterByViewLocale($lang->getLocale())
+                        ->findOneByRedirected(null)
+                    ;
+                    
+                    $domainUrl = $lang->getUrl();
+                    $path = '';
+                    if (null != $urlRewrite) {
+                        $path = "/".$urlRewrite->getUrl();
+                    }
+                    $baseUrl = $request->getBaseUrl();
+                    $url = rtrim($domainUrl, "/").$baseUrl.$path;
+                }
```

Comment:
You can write something like  `$url = rtrim($lang->getUrl(), "/").$request->getBaseUrl().$path;` and save two useless variable affectations

(promise, after this change I'll merge the PR :) )